src/aspire/storage/starfile.py

```diff
@@ -16,24 +14,24 @@ class StarFileError(Exception):
 class StarFile:
     def __init__(self, filepath="", blocks=None):
         """
-        Initialize either from a path to a STAR file or from an OrderedDict of dataframes
+        Initialize either from a path to a STAR file or from an OrderedDict of ndarrays
         """
         # if blocks are given, set self.blocks, otherwise initialize an empty OrderedDict()
+        if blocks is not None:
+            for k in blocks:
+                for _k, _v in blocks[k].items():
+                    if not isinstance(_v, str):
+                        if hasattr(_v, "__iter__"):
```

Comment:
Maybe `try: iter(_v)` is more appropriate here?

Reply:
I will take a look at this a bit more, but I think will result in more code.

Reply:
Only slightly more code, but I added comments so perhaps it is clearer now.

Reply:
This seems to happen often enough in this new code that having a utility `is_iterable` may be appropriate here?

Reply:
Can make an issue.

Reply:
Sure https://github.com/ComputationalCryoEM/ASPIRE-Python/issues/925